### PR TITLE
Automatic releases. Because why not?

### PR DESCRIPTION
**Pull request by Macleykun**
Please don't look at the commit history.
I hope i can squash them later.

### CODE REVIEW

**Comment by airsquared** [comment]
Lol, nice commit history. (Don't worry, it's not as bad as one time I tried experimenting with GitHub Actions where I ended up with like 53 commits)

**Comment by Macleykun** [comment]
So i've tested for a while, and like everything works except the version stuff.
I opend an issue, b/c the env var is outputting the correct value and the action i use just can't.
https://github.com/marvinpinto/actions/issues/178

Perhaps they might advice that what i'm doing isn't needed, and that with using latest etc, the action just checks what the previous version is, and then updates it accordingly (no clue).

Gonna wait what they have to say, after that squash and merge (i believe).

**Comment by airsquared** [comment]
Just with a quick look over, I think to refer to environment variables in a workflow outside of the `run: ...`, you need to use the `env` context. Like `${{ env.GITHUB_VERSION }}` ([source](https://docs.github.com/en/actions/reference/environment-variables#about-environment-variables)).

**Comment by Macleykun** [comment]
> Just with a quick look over, I think to refer to environment variables in a workflow outside of the `run: ...`, you need to use the `env` context. Like `${{ env.GITHUB_VERSION }}` ([source](https://docs.github.com/en/actions/reference/environment-variables#about-environment-variables)).

https://github.com/Macleykun/blobsaver/releases
yep ya both are correct, funny how I read over that haha

anyways, I’m going to bed, any changes ya would like? Feel free to nit pick (squash will happen after that)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -26,3 +26,22 @@ jobs:
...
+    steps:
+      - name: get artefacts
+        uses: actions/download-artifact@v2
+      - name: get version from commit
+        run: echo "GITHUB_VERSION=$(echo ${{ github.event.commits[0].message }} | awk '{print $2}')" >> $GITHUB_ENV 
+      - name: test the var
+        run: echo $GITHUB_VERSION
+      - name: release public builds
+        uses: "marvinpinto/action-automatic-releases@latest"
+        with:
```
Comment (airsquared) [inline .github/workflows/main.yml:43]: Could you please make it create a draft release by default? Before releasing I'd like to modify the description to add more to the changelog and manually hit publish.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: my workflow
```
Comment (airsquared) [inline .github/workflows/test.yml:1]: I think you created this file for testing, could you please delete the file before merging?

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Created by .ignore support plugin (hsz.mobi)
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (airsquared) [inline .gitignore:1]: Guessing this change was for testing?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -26,3 +26,22 @@ jobs:
         with:
           name: ${{ runner.os }}
           path: build/distributions/*
+  release:
+    name: Create pubilc releases
+    runs-on: ubuntu-latest
+    needs: build
+    if: "startsWith(github.event.head_commit.message, 'Release')"
+    steps:
+      - name: get artefacts
```
Comment (airsquared) [inline .github/workflows/main.yml:35]: Minor spelling error, "artifacts"

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -26,3 +26,22 @@ jobs:
...
+        uses: actions/download-artifact@v2
+      - name: get version from commit
+        run: echo "GITHUB_VERSION=$(echo ${{ github.event.commits[0].message }} | awk '{print $2}')" >> $GITHUB_ENV 
+      - name: test the var
```
Comment (airsquared) [inline .github/workflows/main.yml:39]: Same here for testing

**Review by airsquared — CHANGES_REQUESTED** [review]
Yay, thank you! Just a few minor comments to remove the testing code, but other than that looks good!

**Comment by Macleykun** [comment]
Fixed the above, everything looks good here.
Gonna do one final test where i also fill in additional notes :)

**Comment by Macleykun** [comment]
So an interesting note here, is that if you decide to fill in additionel notes, it will break the version part sadly
https://github.com/Macleykun/blobsaver/runs/2697548040?check_suite_focus=true

```
  Searching for SHA corresponding to previous "tags/Release v7.7.7" release tag
  Could not find SHA corresponding to tag "tags/Release v7.7.7" (Not Found). Assuming this is the first release.
  Retrieving commits between HEAD and 12e2e02e0536696fb198edd5bf7050416764a241
  Successfully retrieved 0 commits between HEAD and 12e2e02e0536696fb198edd5bf7050416764a241
Generating changelog
Generating release tag
  Attempting to create or update release tag "Release v7.7.7"
  Could not create new tag "refs/tags/Release v7.7.7" (refs/tags/Release v7.7.7 is not a valid ref name.) therefore updating existing tag "tags/Release v7.7.7"
  Error: Reference does not exist
```
Here it working as we all expect by now:
https://github.com/Macleykun/blobsaver/runs/2697595408?check_suite_focus=true

```
Run echo "VERSION=$(echo Release v7.7.7 | awk '{print $2}')" >> $GITHUB_ENV
  echo "VERSION=$(echo Release v7.7.7 | awk '{print $2}')" >> $GITHUB_ENV
  shell: /usr/bin/bash -e {0}
8s
Run marvinpinto/action-automatic-releases@latest
Initializing the Automatic Releases action
Determining release tags
Retrieving commit history
Generating changelog
Generating release tag
Deleting GitHub releases associated with the tag "v7.7.7"
Generating new GitHub release for the "v7.7.7" tag
  Creating new release
Uploading release artifacts
```

So the pull is technically working, so it's up to you if you find this good enough :). I will squash them in the meanwhile.

**Review by airsquared — APPROVED** [review]

**Comment by airsquared** [comment]
Looks good, thanks again! Actually you don't need to do the squashing yourself, GitHub has a built in "squash and merge" button. I'll be using this to release v3.0 today.

**Comment by airsquared** [comment]
Tried it earlier with v3.0/v3.0.1 and worked perfectly!

**Pull request merged**